tcms/testcases/tests/test_views.py

```diff
@@ -179,6 +179,32 @@ def setUpTestData(cls):
         user_should_have_perm(cls.tester, 'testcases.add_testcase')
         user_should_have_perm(cls.tester, 'testcases.view_testcase')
 
+    def test_notify_formset_invalid(self):
+
+        data = self.data.copy()
+        data['email_settings-0-id'] = -1
+        del data['email_settings-0-auto_to_case_tester']
+
+        response = self.client.post(self.new_case_url, data)
+
+        self.assertContains(response, _('New TestCase'))
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="email_settings-0-auto_to_case_tester" '
+            'type="checkbox"', html=False)
+
+    def test_new_test_case_without_permissions(self):
```

Comment:
Must descend from PermissionsTestCase, see how it is used in other parts of the test suite.

Reply:
I removed this completely cause this edge case was already being tested under test_create_test_case_without_permissions.

Reply:
I don't understand. You say you have removed the scenario "test_new_test_case_without_permissions" but the diff section shows that nothing has been removed. OTOH "test_create_test_case_without_permissions" does exist but it si very old and needs to be updated to inherit from PermissionsTestCase. 

I'm not sure if the PR was old or if something else is going on here.  Can you try rebasing onto the latest master branch so I can investigate.

Reply:
I don't understand. You say you have removed the scenario "test_new_test_case_without_permissions" but the diff section shows that nothing has been removed. OTOH "test_create_test_case_without_permissions" does exist but it si very old and needs to be updated to inherit from PermissionsTestCase. 

I'm not sure if the PR was old or if something else is going on here.  Can you try rebasing onto the latest master branch so I can investigate.

Reply:
Can you please check the latest code? I have created a new class TestNewCasePermission that inherits from PermissionsTestCase.